File: MichiMark/Projection/Common/TagItemProjection.swift
public struct TagItemProjection: Identifiable, Equatable {
    public let id: TagID
    public let tagName: String
    public let isVisible: Bool
    
    init(domain: TagDomain) {
        self.id = domain.id
        self.tagName = domain.tagName
        self.isVisible = domain.isVisible
    }
}

File: flutter/lib/features/shared/projection/tag_item_projection.dart
import 'package:equatable/equatable.dart';

class TagItemProjection extends Equatable {
  final String id;
  final String tagName;
  final bool isVisible;

  const TagItemProjection({
    required this.id,
    required this.tagName,
    required this.isVisible,
  });

  @override
  List<Object?> get props => [id, tagName, isVisible];
}
